semcore/d3-chart/src/RadialTree.tsx

```diff
@@ -0,0 +1,768 @@
+    const data = $rootProps.data[index];
+    const { xLabelCenter, yLabelCenter, labelAngle, isHorizontal } = this.computeRadianPosition(
+      data,
+      index,
+    );
+    const asProps = this.asProps;
+    const { uid, textSize } = asProps;
+    const { label } = data;
+    const color = data.color ?? asProps.color;
+
+    return {
+      x: xLabelCenter,
+      y: yLabelCenter,
+      angle: labelAngle,
+      ['data-radial-animation']: `${uid}-label`,
+      ['data-radian-index']: index,
+      label,
+      color,
+      isHorizontal,
+      textSize,
+    } as IRadialTreeRadianLabelProps;
+  }
+
+  getRadianKey(data: RadianData, index: number) {
+    if (!data) debugger;
+    return data.key ?? `radian-${index}`;
+  }
+
+  computeRadianPosition(data: RadianData, index: number) {
+    const asProps = this.asProps;
+    const { centralMargin, angleOffset, activeKey, size, radiansCount, textWidth } = asProps;
+    const [width, height] = size;
+    const key = this.getRadianKey(data, index);
+    const isActive = activeKey === key;
+    const baseCapSize = data.capSize ?? asProps.capSize;
+    const capSize = baseCapSize * (isActive ? 1 : 0.5);
+
+    const minDemSize = Math.min(width, height) / 2;
+    const length = Math.max(minDemSize - textWidth - baseCapSize - centralMargin, 10);
+
+    const angle = baseAngle + angleOffset + (index / radiansCount) * (Math.PI * 2);
+    const isHorizontal =
+      (angle - baseAngle > (1 / 6) * Math.PI && angle - baseAngle < (5 / 6) * Math.PI) ||
+      (angle - baseAngle > (7 / 6) * Math.PI && angle - baseAngle < (11 / 6) * Math.PI);
+    const topAngle = -Math.PI / 2;
+    const labelAngle = ((angle - topAngle) % Math.PI) + topAngle;
+
+    const center = [width / 2, height / 2];
+    const [xCenter, yCenter] = center;
+    const start = [
+      xCenter + Math.cos(angle) * centralMargin,
+      yCenter + Math.sin(angle) * centralMargin,
+    ];
+    const end = [
+      xCenter + Math.cos(angle) * (centralMargin + length),
+      yCenter + Math.sin(angle) * (centralMargin + length),
+    ];
+    const [xStart, yStart] = start;
+    const [xEnd, yEnd] = end;
+
+    const labelCenter = [
+      xCenter + Math.cos(angle) * (centralMargin + length + baseCapSize + textWidth / 2),
+      yCenter + Math.sin(angle) * (centralMargin + length + baseCapSize + textWidth / 2),
+    ];
+    const [xLabelCenter, yLabelCenter] = labelCenter;
+
+    return {
+      xStart,
+      yStart,
+      xEnd,
+      yEnd,
+      xLabelCenter,
+      yLabelCenter,
+      labelAngle,
+      isHorizontal,
+      capSize,
+      isActive,
+    };
+  }
+
+  renderRadian(data: RadianData, index: number) {
+    const asProps = this.asProps;
+    const { styles, Children, onRadianClick } = asProps;
+    const key = this.getRadianKey(data, index);
+    const SRadian = 'g';
+
+    return sstyled(styles)(
+      <SRadian key={key} onClick={onRadianClick(key)}>
+        <Children />
```

Comment:
How I understand, if I write this `<RadialTree data={array length 5}>test</RadialTree>` I get text "test" 5 times. I think this mustn't that work.

Reply:
With a little change to: `<RadialTree data={%%array on n-th size%%}><RadialTree.Radian>test</RadialTree.Radian></RadialTree>` it really will render "test" n times (but text will not be displayed as far as it is plain text inside of `<svg />` element). Totally the same like in bar chart